DiscordChatExporter.Domain/Discord/Models/Channel.cs

```diff
@@ -36,44 +37,67 @@ public partial class Channel : IHasId
 
         public Snowflake GuildId { get; }
 
-        public string Category { get; }
+        public Channel Category { get; }
 
         public string Name { get; }
 
+        public int Position { get; }
+
         public string? Topic { get; }
 
-        public Channel(Snowflake id, ChannelType type, Snowflake guildId, string category, string name, string? topic)
+        public Channel(Snowflake id, ChannelType type, Snowflake guildId, Channel? category, string name, int position, string? topic)
         {
             Id = id;
             Type = type;
             GuildId = guildId;
-            Category = category;
+            Category = category ?? (type != ChannelType.GuildCategory ? GetDefaultCategory(type, guildId) : this);
```

Comment:
That's a good point, I didn't think of this.

I can see a couple of solutions to this:

1. Keep `Category` nullable, but then we have to be a bit more defensive in the output writers, as we can't rely on the category being set.
2. Create a `ChannelCategory` class, which is a `Channel` with only properties relevant to a category. So it wouldn't have a parent category of its own (IIRC that's not possible, right?) and topic.

And finally, the greatest of them all:

3. Do we really need the expanded category metadata? It seems what we're using it for is: category position and category ID. As a Discord user, I think, you're not even aware that the categories are actual channels rather than simple labels, so perhaps those are not very useful. I personally have a hard time imagining where I would want the category ID or position to be included. What do you think? If we can drop it, we can revert the changes to category parsing and make it significantly simpler.
3.

Reply:
Approach `3.` may break the GUI sorting though, because then we wouldn't be able to sort the categories by position.

Just to clarify, `Channel.Position` is the channels global position within the guild, or is it within its category?

Reply:
I think you raise a valid point in that I was definitely attaching more data than needed. I think condensing the data into a few properties on the `Channel` as in approach `3.` is fine, unless it becomes too cluttered in which case it may be better to follow solution `2.` instead.

I think position is definitely a useful piece of data, since it lets you organize categories and channels in your filesystem the same way which they would appear in Discord. For example, I currently have a script that exports my servers like so:
![image](https://user-images.githubusercontent.com/21089343/105786364-ed657500-5f4a-11eb-8e81-88c9e383f2fe.png)
ID is perhaps not as useful, but I could see users potentially wanting to access that.

As for the actual value of `Channel.Position`...well, I originally intended it to be its position within the category, but Discord seemed to return global position through the API so I just went with that. I've done some more digging since making the initial PR proposal and it seems that the `position` value which Discord attaches to each channel isn't absolute, without any guarantees that the positions will be unique or contiguous. It's not very clear on the documentation, but there's some discussion about it [here](https://github.com/Rapptz/discord.py/issues/2392). As such, using Discord's raw response may not actually be useful at all. I think it's definitely possible to assign position values (both within category and global) when enumerating the channels, but I believe that would result in an extra request for the CLI if exporting a single channel (though, in that case, maybe channel position isn't very important). However, things might get a bit messy if we stuff in manual data that can't be parsed with `Channel.Parse`. Thoughts?

Reply:
I see, category position does appear useful.
Maybe we can create a special type, `ChannelCategory` which would just contain the name, the position, and the ID? I don't know off the top of my head if that helps simplify things, but at least there will be no referential recursion.

Reply:
Sounds good. Since the type would overlap with `Channel`, do you think it would be better to use some sort of inheritance or keep them completely separate?

Reply:
Let's keep them completely separate to leave our options open for the future.